src/AstTransforms.cpp

```diff
@@ -347,14 +346,18 @@ bool MaterializeAggregationQueriesTransformer::materializeAggregationQueries(
                     visitDepthFirst(arg, [&](const AstVariable& var) { varCtr[var.getName()]++; });
                 }
             });
-            for (size_t i = 0; i < aggAtom->getArity(); i++) {
-                if (auto* var = dynamic_cast<AstVariable*>(aggAtom->getArgument(i))) {
+            std::vector<AstArgument*> args;
+            for (auto arg : head->getArguments()) {
+                if (auto* var = dynamic_cast<AstVariable*>(arg)) {
                     // replace local variable by underscore if local
                     if (varCtr[var->getName()] == 0) {
-                        aggAtom->setArgument(i, std::make_unique<AstUnnamedVariable>());
+                        args.push_back(new AstUnnamedVariable());
+                        continue;
                     }
                 }
+                args.push_back(arg);
             }
+            AstAtom* aggAtom = new AstAtom(*head, args);
```

Comment:
The new atom and the old atom both store unique pointers to the same arguments. You will need to clone the copied arguments.